Tests/L0/Maven/_suite.ts

```diff
@@ -44,58 +86,108 @@ describe('Maven Suite', function() {
+        responseJsonContent = setupMockResponsesForPaths(responseJsonContent,
+            [testXmlFilePath, testHtmlFilePath, agentStgDir, codeAnalysisStgDir, pmdStgDir, moduleStgDir]);
+        // Test data files
+        responseJsonContent = setupMockResponsesForPaths(responseJsonContent, listFolderContents(agentSrcDir));
 
-        responseJsonContent.checkPath = responseJsonContent.checkPath || {};
-        responseJsonContent.checkPath[agentStgDir] = true;
-        responseJsonContent.checkPath[codeAnalysisStgDir] = true;
-        responseJsonContent.checkPath[pmdStgDir] = true;
-        responseJsonContent.checkPath[testXmlFilePath] = true;
-        responseJsonContent.checkPath[testHtmlFilePath] = true;
+        // Write and set the newly-changed response file
+        var newResponseFilePath:string = path.join(agentStgDir, 'response.json');
+        fs.writeFileSync(newResponseFilePath, JSON.stringify(responseJsonContent));
+        setResponseFile(newResponseFilePath);
+
+        // Set up the task runner with the test settings
+        var taskRunner:tr.TaskRunner = setupDefaultMavenTaskRunner();
+        taskRunner.setInput('pmdAnalysisEnabled', 'true');
+        taskRunner.setInput('test.artifactStagingDirectory', agentStgDir);
+        taskRunner.setInput('test.sourcesDirectory', agentSrcDir);
+
+        // Act
+        taskRunner.run()
+            .then(() => {
+
+                // Assert
+                assert(taskRunner.resultWasSet, 'should have set a result');
+                assert(taskRunner.stdout.length > 0, 'should have written to stdout');
+                assert(taskRunner.succeeded, 'task should have succeeded');
+
+                assert(taskRunner.ran('/usr/local/bin/mvn -f pom.xml package pmd:pmd -DlinkXRef=false'),
+                    'should have run maven with the correct arguments');
+                assert(taskRunner.stdout.indexOf('task.addattachment type=Distributedtask.Core.Summary;name=Code Analysis Report') > -1,
+                    'should have uploaded a Code Analysis Report build summary');
+
+                assert(taskRunner.stdout.indexOf('artifact.upload containerfolder=singlemodule;artifactname=') > -1,
+                    'should have uploaded PMD build artifacts');
 
-        responseJsonContent.rmRF = responseJsonContent.rmRF || {};
-        var rmRFSuccessObj = {
-            success: true,
-            message: "foo bar"
-        };
-        responseJsonContent.rmRF[agentStgDir] = rmRFSuccessObj;
-        responseJsonContent.rmRF[codeAnalysisStgDir] = rmRFSuccessObj;
-        responseJsonContent.rmRF[pmdStgDir] = rmRFSuccessObj;
+                done();
+            })
+            .fail((err) => {
+                console.log(taskRunner.stdout);
+                console.log(taskRunner.stderr);
+                console.log(err);
+                done(err);
+            });
+    });
+
+    it('Maven / PMD: Detects and uploads results when multiple modules are present', (done) => {
```

Comment:
The content of the uploaded build summary file is not checked by the tests.

Accepted.